Sources/MeiliSearch/Error.swift

```diff
@@ -21,9 +66,45 @@ public extension MeiliSearch {
 
     /// The input or output JSON is invalid.
     case invalidJSON
-  }
 
-}
+    // URL is invalid
+    case invalidURL
+
+    /// Error originating from MeiliSearch API.
+    // case meiliSearchApiError(message: String, errorCode: String, errorType: String, errorLink: String? = "http://docs.meilisearch.com/errors", underlying: Swift.Error)
+    case meiliSearchApiError(message: String, msErrorResponse: MSErrorResponse?, statusCode: Int = 0, url: String = "")
+
+    /// Error communicating with MeiliSearch API.
+    case meiliSearchCommunicationError(message: String, url: String)
 
-/// Allow use of error comparasion for the MeiliSearch.Error type.
-extension MeiliSearch.Error: Equatable {}
+    public var errorDescription: String? {
+      switch self {
+      case .serverNotFound:
+        return "All hosts are unreachable."
+      case .dataNotFound:
+        return "Missing response data"
+      case .hostNotValid:
+        return "Response decoding failed"
+      case .invalidJSON:
+        return "Invalid json"
+      case .invalidURL:
+        return "Invalid host URL"
+      case .meiliSearchCommunicationError(let message, let url):
+        return "meiliSearchCommunicationError \(message) \(url) "
+
+      // can we do let error and then error.message etc...
+      case .meiliSearchApiError(let message, let error, let statusCode, let url):
+        if let msErrorResponse = error as? MSErrorResponse {
```

Comment:
```suggestion
        if let msErrorResponse = error as MSErrorResponse? {
```
I'm not sure but I think this is more accurate